cclib/parser/logfileparser.py

```diff
@@ -200,35 +51,22 @@ def __init__(self, source: Union[str, Iterable[str], fileinput.FileInput], logle
             logstream - where to output the logging information
             datatype - class to use for gathering data attributes
         """
-
-        # Set the filename to source if it is a string or a list of strings, which are
-        # assumed to be filenames. Otherwise, assume the source is a file-like object
-        # if it has a read method, and we will try to use it like a stream.
-        self.isfileinput = False
-        if isinstance(source, str):
-            self.filename = source
-            self.isstream = False
-        elif isinstance(source, list) and all([isinstance(s, str) for s in source]):
-            self.filename = source
-            self.isstream = False
-        elif isinstance(source, fileinput.FileInput):
-            self.filename = source
-            self.isstream = False
-            self.isfileinput = True
-        elif hasattr(source, "read"):
-            self.filename = f"stream {str(type(source))}"
-            self.isstream = True
-            self.stream = source
-        else:
-            raise ValueError("Unexpected source type.")
+        if not isinstance(source, FileWrapper):
+            source = FileWrapper(source)
+            # Probably the wrong type given.
+            #raise TypeError("Source does not have an 'input_files' attribute, are you sure it inherits from FileWrapper?") from None
```

Comment:
Is it safe to get rid of this?

Reply:
Should be aye, the comment is from an earlier version of mine. Current implementation in master is this:
```
        # Set the filename to source if it is a string or a list of strings, which are
        # assumed to be filenames. Otherwise, assume the source is a file-like object
        # if it has a read method, and we will try to use it like a stream.
        self.isfileinput = False
        if isinstance(source, str):
            self.filename = source
            self.isstream = False
        elif isinstance(source, list) and all([isinstance(s, str) for s in source]):
            self.filename = source
            self.isstream = False
        elif isinstance(source, fileinput.FileInput):
            self.filename = source
            self.isstream = False
            self.isfileinput = True
        elif hasattr(source, "read"):
            self.filename = f"stream {str(type(source))}"
            self.isstream = True
            self.stream = source
        else:
            raise ValueError("Unexpected source type.")
```
All this type juggling is now done by FileWrapper, if the source isn't FileWrapper we simple make it one and all should be dandy.